client/src/session_manager.rs

```diff
@@ -721,6 +794,46 @@ impl SessionManager {
         Ok(session)
     }
 
+    async fn assert_remote_node(
+        &self,
+        node_id: NodeId,
+        addrs: &[SocketAddr],
+    ) -> anyhow::Result<()> {
+        // Match against own node id
+        if node_id == self.config.node_id {
+            anyhow::bail!("remote id belongs to this node");
+        }
+
+        // Match against own IP addresses
+        let own_addrs: Vec<_> = {
+            let state = self.state.read().await;
+            vec![state.bind_addr, state.public_addr]
+                .into_iter()
+                .flatten()
+                .collect()
+        };
+        for addr in addrs {
+            if own_addrs.iter().any(|a| a == addr) {
```

Comment:
I'm not sure of this. I always assumed, that this addresses list can contain public IP, but it can also contain private IPs. For example 2 Nodes in the same network could communicate directly without needing to route packets through all routers, but only the closest one.
In this case these addresses can overlap between Nodes.
But of course we don't have this situation right now...

Reply:
I've taken a different approach, where we remove our own addresses from `addrs`

Reply:
> For example 2 Nodes in the same network could communicate directly without needing to route packets through all routers, but only the closest one.
> In this case these addresses can overlap between Nodes.

In general, this is not the best way to manage nodes in a local network - we're relying on trial and error. Local announcements via mdns / broadcasts would allow the node to build a reliable knowledge base of reachable nodes